Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs

```diff
@@ -0,0 +1,152 @@
+            if (!queryResult.CampaignHasContent)
+            {
+                return new EmptyContentData(campaignId);
+            };
+
+            // TODO: test this scenario
+            if (queryResult.EditorType == EDITOR_TYPE_MSEDITOR)
+            {
+                return new MSEditorContentData(campaignId, queryResult.Content);
+            }
+
+            // TODO: test this scenario
+            // Related tests:
+            // * GET_campaign_should_accept_right_tokens_and_return_unlayer_content
+            if (queryResult.EditorType == EDITOR_TYPE_UNLAYER)
+            {
+                return new UnlayerContentData(
+                    campaignId: queryResult.IdCampaign,
+                    htmlContent: queryResult.Content,
+                    meta: queryResult.Meta);
+            }
+
+            // TODO: test this scenario
+            // Related tests:
+            // * GET_campaign_should_return_html_content
+            if (queryResult.EditorType == null)
+            {
+                return new HtmlContentData(
+                    campaignId: queryResult.IdCampaign,
+                    htmlContent: queryResult.Content);
+            }
+
+            // TODO: test this scenario
+            return new UnknownContentData(
+                campaignId: queryResult.IdCampaign,
+                content: queryResult.Content,
+                meta: queryResult.Meta,
+                editorType: queryResult.EditorType);
+        }
+    }
+
+    public async Task SaveCampaignContent(string accountName, ContentData contentRow)
+    {
+        using (var connection = await _connectionFactory.GetConnection())
+        {
+            var databaseQuery = @"
+SELECT
+CAST (CASE WHEN ca.IdCampaign IS NULL THEN 0 ELSE 1 END AS BIT) AS OwnCampaignExists,
+CAST (CASE WHEN co.IdCampaign IS NULL THEN 0 ELSE 1 END AS BIT) AS ContentExists,
+co.EditorType
+FROM [User] u
+LEFT JOIN [Campaign] ca ON u.IdUser = ca.IdUser
+AND ca.IdCampaign = @IdCampaign
+LEFT JOIN [Content] co ON ca.IdCampaign = co.IdCampaign
+WHERE u.Email = @accountName
+";
+            // TODO: use a type for the result
+            var campaignStatus = await connection.QueryFirstOrDefaultAsync<dynamic>(databaseQuery, new { contentRow.campaignId, accountName });
+
+            // TODO: consider returning 404 NotFound
+            // TODO: test this scenario
+            if (!campaignStatus.OwnCampaignExists)
+            {
+                throw new ApplicationException($"CampaignId {contentRow.campaignId} does not exists or belongs to another user than {accountName}");
+            }
+
+            // TODO: test these scenarios
+            var query = campaignStatus.ContentExists
+                ? @"UPDATE Content SET Content = @Content, Meta = @Meta, EditorType = @EditorType WHERE IdCampaign = @IdCampaign"
+                : @"INSERT INTO Content (IdCampaign, Content, Meta, EditorType) VALUES (@IdCampaign, @Content, @Meta, @EditorType)";
```

Comment:
I think that It is not necessary 
```suggestion
                ? "UPDATE Content SET Content = @Content, Meta = @Meta, EditorType = @EditorType WHERE IdCampaign = @IdCampaign"
                : "INSERT INTO Content (IdCampaign, Content, Meta, EditorType) VALUES (@IdCampaign, @Content, @Meta, @EditorType)";
```